src/library/scanner/libraryscanner.cpp

```diff
@@ -170,6 +156,17 @@ void LibraryScanner::run() {
 
 void LibraryScanner::slotStartScan() {
     qDebug() << "LibraryScanner::slotStartScan";
+    DEBUG_ASSERT(m_state == STARTING);
```

Comment:
please use debug_assert_and_handle so we don't crash.  maybe just return if this invariant doesn't hold